src/BrainPass/BrainPass.sol

```diff
@@ -0,0 +1,402 @@
+
+        (bool success, ) = (msg.sender).call{value: ethbalance}("");
+        if (!success) revert TransferFailed();
+    }
+
+    function withdrawIQ() external payable onlyOwner {
+        uint256 tokenBalance = IERC20(iqToken).balanceOf(address(this));
+
+        if (tokenBalance <= 0) revert NoIQLeftToWithdraw();
+
+        bool tokenSuccess = IERC20(iqToken).transfer(msg.sender, tokenBalance);
+        if (!tokenSuccess) revert TransferFailed();
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Internal Functions
+    /// -----------------------------------------------------------------------
+
+    /// @notice Calculate the price of an Nft
+    /// @param startTimestamp The start time to calculate the price of the Nft
+    /// @param endTimestamp The end time to calculate the price of the Nft
+    function calculatePrice(
+        uint256 passId,
+        uint256 startTimestamp,
+        uint256 endTimestamp
+    ) internal view returns (uint256) {
+        PassType memory passType = passTypes[passId];
+        uint256 subscriptionPeriodInSeconds = endTimestamp - startTimestamp;
+
+        uint256 subscriptionPeriodInDays = subscriptionPeriodInSeconds / 1 days;
+
+        // Calculate the total price
+        uint256 totalPrice = subscriptionPeriodInDays * passType.pricePerDay;
+
+        if (passType.discount > 0) {
+            uint256 discountAmount = (totalPrice * passType.discount) / (100);
+            totalPrice = totalPrice - discountAmount;
+        }
+
+        return totalPrice;
+    }
+
+    /// @notice Validates the Timestamp Duration for minting Nft
+    /// @param startTimestamp The start time for checking the validity of a pass
+    /// @param endTimestamp The end time for checking the validity of a pass
+    function validatePassDuration(
+        uint256 startTimestamp,
+        uint256 endTimestamp
+    ) internal view returns (bool) {
+        uint256 durationInDays = (endTimestamp - startTimestamp) / 1 days;
+        if (endTimestamp < block.timestamp) {
+            revert DurationNotInTimeFrame();
+        }
+        return
+            durationInDays >= MINT_LOWER_LIMIT &&
+            durationInDays <= MINT_UPPER_LIMIT;
+    }
+
+    /// -----------------------------------------------------------------------
+    /// Getters
+    /// -----------------------------------------------------------------------
+
+    /// @notice Gets the tokenUri for the contract
+    function _baseURI() internal view virtual override returns (string memory) {
+        return baseTokenURI;
+    }
+
+    /// @notice Gets all the NFT owned by an address
+    /// @param user The address of the user
+    function getUserPassDetails(
+        address user
+    ) public view returns (UserPassItem memory) {
+        UserPassItem memory userToken = addressToNFTPass[user];
+        PassType storage passType = passTypes[userToken.passId];
+        if (passType.isPaused) revert PassTypeIsPaused();
```

Comment:
i dont think you need to revert a view if a passtype is paused. paused its just for minting